### PR TITLE
[Nightly test] Make report False by default.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Currently, the release test result is reported to our database by default. This corrupted the nightly dashboard when some people manually invoke nightly tests because the result is reported without modifying RAY_BRANCH env variable.

This PR makes this better by making --no-report as a default behavior. If we'd like to report the result, we should specify REPORT_RESULT=1 as an environment variable.

Please do not merge this until all buildkite schedule has REPORT_RESULT=1 env vars.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]

**Comment by rkooo567** [comment]
Will merge once Kai also approves it. Please do not merge until I do that.

**Review by krfricke — APPROVED** [review]
Thanks, this is great!

**Comment by rkooo567** [comment]
Test failure should be unrelated. All nightly tests config has updated to include `REPORT_RESULT=1`.

**Pull request merged**
